eth/blockwatch/block_watcher.go

```diff
@@ -281,28 +284,45 @@ func (w *Watcher) addLogs(header *MiniHeader) (*MiniHeader, error) {
 func (w *Watcher) getMissedEventsToBackfill(ctx context.Context) ([]*Event, error) {
 	events := []*Event{}
 
+	var (
+		startBlockNum          int
+		latestRetainedBlockNum int
+		blocksElapsed          int
+	)
+
 	latestRetainedBlock, err := w.stack.Peek()
 	if err != nil {
 		return events, err
 	}
-	// No blocks stored, nowhere to backfill to
-	if latestRetainedBlock == nil {
-		return events, nil
-	}
+
 	latestBlock, err := w.client.HeaderByNumber(nil)
 	if err != nil {
 		return events, err
 	}
-	blocksElapsed := big.NewInt(0).Sub(latestBlock.Number, latestRetainedBlock.Number)
-	if blocksElapsed.Int64() == 0 {
+	latestBlockNum := int(latestBlock.Number.Int64())
+
+	if latestRetainedBlock != nil {
+		latestRetainedBlockNum = int(latestRetainedBlock.Number.Int64())
+		// Events for latestRetainedBlock already processed, start at latestRetainedBlock + 1
+		startBlockNum = latestRetainedBlockNum + 1
+		blocksElapsed = latestBlockNum - startBlockNum
+	} else if w.backfillStartBlock != nil {
+		startBlockNum = int(w.backfillStartBlock.Int64())
+		blocksElapsed = latestBlockNum - startBlockNum
+	} else {
 		return events, nil
 	}
 
-	glog.Infof("Some blocks have elapsed since last boot. Backfilling block events (this can take a while)... blocksElapsed=%v", blocksElapsed.Int64())
-	startBlockNum := int(latestRetainedBlock.Number.Int64() + 1)
-	endBlockNum := int(latestRetainedBlock.Number.Int64() + blocksElapsed.Int64())
+	if blocksElapsed == 0 {
+		return events, nil
+	}
+
+	glog.Infof("Some blocks have elapsed since last boot. Backfilling block events (this can take a while)...\n")
```

Comment:
In the case where we're backfilling from `backfillStartBlock` then there was no "last boot". I think we can just remove the first part of this statement and just log "Backfilling block events...<rest of log>"

Reply:
fixed in be50d19d